lib/stores/DataStore.js

```diff
@@ -37,13 +41,20 @@ class DataStore {
      *
      * http://tus.io/protocols/resumable-upload.html#creation
      *
-     * @param  {object} file File model.
+     * @param  {object} req http.incomingMessage
+     * @return {Promise}
      */
-    create(file) {
-        if (!(file instanceof File)) {
-            throw new Error(`${file} is not a File`);
-        }
-        console.log(`[DataStore] create: ${file.id}`);
+    create(req) {
+        console.log('[DataStore] create');
+
+        return new Promise((resolve, reject) => {
```

Comment:
Right, since the whole point of `create` is to create an instance of `File` with all the properties about its location, etc, it makes more sense to return the whole `File` rather than just the name. Edits coming soon!